src/query/functions-v2/src/scalars/comparison.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_expression::types::BooleanType;
+use common_expression::types::NumberDataType;
+use common_expression::types::NumberType;
+use common_expression::types::StringType;
+use common_expression::types::ALL_NUMERICS_TYPES;
+use common_expression::with_number_mapped_type;
+use common_expression::FunctionProperty;
+use common_expression::FunctionRegistry;
+use regex::bytes::Regex as BytesRegex;
+use regex::bytes::RegexBuilder as BytesRegexBuilder;
+
+pub fn register(registry: &mut FunctionRegistry) {
+    registry.register_2_arg::<StringType, StringType, BooleanType, _, _>(
```

Comment:
BooleanType, TimestampType are not covered.

Can use macro to generate the codes like `ALL_NUMERICS_TYPES`

Reply:
Got